src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -340,112 +345,68 @@ class PostgresEntityDataQueryService(
-                        upsertPropertyValue.first.setInt(8, partitionsVersion)
-                        upsertPropertyValue.first.setObject(9, insertValue)
-                        upsertPropertyValue.first.addBatch()
-
-                        val maybeLinkingId = entityKeyIdsToLinkingIds[entityKeyId]
-                        if (maybeLinkingId != null) {
-                            // update for linked rows
-                            upsertPropertyValue.second.setObject(1, entitySetId)
-                            upsertPropertyValue.second.setObject(2, maybeLinkingId)
-                            upsertPropertyValue.second.setInt(3, getPartition(maybeLinkingId, allPartitions))
-                            upsertPropertyValue.second.setObject(4, propertyTypeId)
-                            upsertPropertyValue.second.setObject(5, propertyHash)
-                            upsertPropertyValue.second.setObject(6, version)
-                            upsertPropertyValue.second.setArray(7, versionsArrays)
-                            upsertPropertyValue.second.setInt(8, partitionsVersion)
-                            upsertPropertyValue.second.setObject(9, insertValue)
-                            upsertPropertyValue.second.setObject(10, entityKeyId)
-                            upsertPropertyValue.second.addBatch()
-                        }
+                        upsertPropertyValue.setObject(1, entitySetId)
+                        upsertPropertyValue.setObject(2, entityKeyId)
+                        upsertPropertyValue.setInt(3, partition)
+                        upsertPropertyValue.setObject(4, propertyTypeId)
+                        upsertPropertyValue.setObject(5, propertyHash)
+                        upsertPropertyValue.setObject(6, version)
+                        upsertPropertyValue.setArray(7, versionsArrays)
+                        upsertPropertyValue.setInt(8, partitionsVersion)
+                        upsertPropertyValue.setObject(9, insertValue)
+                        upsertPropertyValue.addBatch()
                     }
                 }
-                upsertPropertyValues.values.map { (entity, link) ->
-                    entity.executeBatch().sum() + link.executeBatch().sum()
-                }.sum()
+                upsertPropertyValues.values.map { it.executeBatch().sum() }.sum()
             }.sum()
-            updatedEntityCount to updatedPropertyCounts
+
+            //Make data visible by marking new version in ids table.
+            val upsertEntities = connection.prepareStatement(buildUpsertEntitiesAndLinkedData())
+
+            upsertEntities.setObject(1, versionsArrays)
+            upsertEntities.setObject(2, version)
+            upsertEntities.setObject(3, version)
+            upsertEntities.setObject(4, entitySetId)
+            upsertEntities.setArray(5, entityKeyIdsArr)
+            upsertEntities.setInt(6, partition)
```

Comment:
I'll fix this after get back from Sacramento. Will require a second CTE query to compute the partitions of all the linking ids.

Reply:
Never mind. This should have been done this way from the beginning. I vaguely recall discussing this and wanting to use the same partition since it co-locates linking related operations.

Reply:
(a) This was easier than expected and actually ended up removing that weird read before write. (b) delete doesn't have an implementation that removes linked entities, should be a separate PR/bugfix. (c) We're still doing two statements tombstones, we should switch to one statement tombstones but that is a separate ticket.

Reply:
(c) https://jira.openlattice.com/browse/LATTICE-2281

Reply:
This will also require changes in PostgresLinkingQueryService

Reply:
+1